grails-app/controllers/org/pih/warehouse/api/FulfillmentApiController.groovy

```diff
@@ -0,0 +1,59 @@
+package org.pih.warehouse.api
+
+import grails.converters.JSON
+import grails.util.Holders
+import org.grails.plugins.web.taglib.ApplicationTagLib
+import org.pih.warehouse.CommandUtils
+import org.pih.warehouse.fulfillment.FulfillmentService
+import org.pih.warehouse.outbound.ImportPackingListCommand
+import org.pih.warehouse.outbound.ImportPackingListErrors
+import org.pih.warehouse.outbound.ImportPackingListItem
+
+class FulfillmentApiController {
+
+    FulfillmentService fulfillmentService
+
+    def validate(ImportPackingListCommand command) {
+        ImportPackingListErrors errors = new ImportPackingListErrors()
+        if (command.hasErrors()) {
+            response.status = 400
+            buildErrors(command, errors)
+        }
+        List<Map> tableData = command.packingList.collect { ImportPackingListItem item -> item.toTableJson() }
+        render([errors: errors, data: tableData] as JSON)
```

Comment:
The table json is needed due to some requirements provided by Kasia and the table itself. 
As you know, we bind domains via id, so for example when sending packingList, we provide product ids, so the payload looks like:
```json
{
  packingList: [
    { product: 10001, ...},
    { product; 10002, ...},
  ]
}
```
so if we were about to rely on this payload to display the data in the table, we would not be able to display the product name, this is where the `toTableJson` is needed - since we bound the product at the bind time, we return it to the client, so that we can display the: product name, bin location name, expiration date associated with the inventory item in the table:
```groovy
[
        rowId: rowId,
        product: [
            id: product?.id,
            name: product?.name,
            productCode: product?.productCode
        ],
        binLocation: binLocation?.name,
        palletName: palletName,
        boxName: boxName,
        lotNumber: lotNumber,
        quantityPicked: quantityPicked,
        recipient: recipient?.name,
        expirationDate: expirationDateToDisplay,
        origin: [
            id: origin?.id,
        ],
]
```